h2o-docs/src/product/productionizing.rst

```diff
@@ -168,6 +168,93 @@ The examples below describe how to start H2O and create a model using R and Pyth
 	    Has penetrated the prostatic capsule (1 yes; 0 no): 0
 	    Class probabilities: 0.8059929056296662,0.19400709437033375
 
+Building a MOJO from sparkling water
+''''''''''''''''''''''''''''''''''''
+ When using scala with sparkling-water, a slightly different approach has to be taken as sparkling water does not generate a zip file.
+ This approach will write the bytestream to a file, which then can be read back from an application without sparkling water.
+
+ **Step 1: Build and extract a model **:
+
+ 1. Open a terminal window and start sparkling-water.
+ 2. Run the following commands to build a simple GBM model.
+
+   .. code:: scala
+
+       implicit val hc = H2OContext.getOrCreate(spark)
```

Comment:
Please see the comment on the PR. I think that this code should be:
 - simplified
 - make sure it is runnable
 - use the exportMOJOModel in the ModelSerializationSupport which can already export to the desired mojo.zip file
- load the mojo using the public `MojoModel.load` method.

Proposal is here
```

// Prepare the environment
import org.apache.spark.h2o._
import hex.genmodel.utils.DistributionFamily
import hex.tree.gbm.GBMModel.GBMParameters
import hex.tree.gbm.GBM
val hc = H2OContext.getOrCreate(spark)

// Load and preapre the data
val table: H2OFrame = new H2OFrame(new java.io.File("examples/smalldata/prostate/prostate.csv"))
val target = "CAPSULE"
table.replace(table.find(target), table.vec(target).toCategoricalVec).remove()

// Build GBM model
val gbmParams = new GBMParameters()
gbmParams._train = table._key
gbmParams._response_column = target
gbmParams._ntrees = 5
gbmParams._nfolds = 3
gbmParams._min_rows = 10
gbmParams._distribution = DistributionFamily.multinomial
val gbm = new GBM(gbmParams)
val model = gbm.trainModel.get

// Export the MOJO
import water.support.ModelSerializationSupport._
exportMOJOModel(model, "mojo.zip", false)



// Load the MOJO
import hex.genmodel.easy.{EasyPredictModelWrapper, RowData}
import hex.genmodel.MojoModel
val mojoModel = MojoModel.load("mojo.zip")

val config = new EasyPredictModelWrapper.Config()
config.setModel(mojoModel)
config.setConvertUnknownCategoricalLevelsToNa(true)

val model = new EasyPredictModelWrapper(config)
val row = new RowData
row.put("AGE", "68")
row.put("RACE", "2")
row.put("DCAPS", "2")
row.put("VOL", "0")
row.put("GLEASON", "6")
val p = model.predictBinomial(row)


println("Has penetrated the prostatic capsule (1=yes; 0=no): " + p.label)
println("Class probabilities: " + p.classProbabilities.mkString(", "))
```

Reply:
Sorry but i don't agree. Even though updating the writing part to utilize ModelSerializationSupport's function writes to a file named zip, the `MojoModel.load()` function fails with the following exception:
error in opening zip file
java.util.zip.ZipException: error in opening zip file
	at java.util.zip.ZipFile.open(Native Method)
	at java.util.zip.ZipFile.<init>(ZipFile.java:220)
	at java.util.zip.ZipFile.<init>(ZipFile.java:150)
	at java.util.zip.ZipFile.<init>(ZipFile.java:121)
	at hex.genmodel.ZipfileMojoReaderBackend.<init>(ZipfileMojoReaderBackend.java:13)
	at hex.genmodel.MojoModel.load(MojoModel.java:33)

I've tried different approaches, and documented them all in my stackoverflowpost https://stackoverflow.com/questions/49515618/how-to-export-an-h2o-model-as-mojo-from-sparkling-water-in-scala-to-be-loaded-b/49521432#49521432.

Reply:
@gerbenoostra I just tried that, please follow the code I created. Verified on Sparkling Water 2.2.10 and works fine for me. I closed your PR in the SW as it's just duplicated already existing methods

Reply:
ps , I did update the code example with the other proposals